tests/fixtures/random_allocator.h

```diff
@@ -0,0 +1,69 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <random>
+
+#include "vsag/allocator.h"
+
+namespace fixtures {
+
+class RandomAllocator : public vsag::Allocator {
+public:
+    std::string
+    Name() override {
+        return "random_allocator";
+    }
+    RandomAllocator() {
+        rd_ = std::make_shared<std::random_device>();
+        gen_ = std::make_shared<std::mt19937>((*rd_)());
+        std::uniform_int_distribution<int> seed_random;
+        std::uniform_real_distribution<float> ratio_range(0.001f, 0.1f);
+        int seed = seed_random(*gen_);
+        float r = ratio_range(*gen_);
+        std::cout << "seed: " << seed << "   error ratio: " << r << std::endl;
+        gen_->seed(seed);
+        error_ratio_ = r;
```

Comment:
Using a fixed constant for `error_ratio_` would be more reliable/definite in identifying problems

Reply:
done